sympy/functions/combinatorial/factorials.py

```diff
@@ -861,8 +861,9 @@ def eval(cls, n, k):
                 return S.Zero
             if n.is_integer and n.is_nonnegative and d.is_negative:
                 return S.Zero
-            if n.is_Number:
-                return cls._eval(n, k)
+            if n.is_number:
+                from sympy.simplify.simplify import simplify
+                return simplify(cls._eval(n, k))
```

Comment:
Could some special part of `simplify` be used instead of the whole sequence?

Reply:
I didn't understand this. Do you mean using any another specific function that would simplify or simplifying only a particular part of the result?

Reply:
If you look at the code of `simplify`, you will find that it is a sequence of different types of simplifications: factor_terms, hyperexpand, piecewise_fold, etc. In many cases only a particular type of simplification need be applied.

Reply:
I looked through the code, tried few simplifications but didn't work.
Since we need to simplify only for cases like this: `binomial((2*I + 3), 7))`, the result of such types is of the form `(P.O.S)/n`, like: `I*(-3 + 2*I)*(-2 + 2*I)*(-1 + 2*I)*(1 + 2*I)*(2 + 2*I)*(3 + 2*I)/2520`, is there any function for this

Reply:
`.expand()` (with optional keywords) will often work.